### PR TITLE
ethernet: T2891: Add ethernet ring-buffer CLI commands

### PR DESCRIPTION
**Pull request by DmitriyEshenko**

### CODE REVIEW

File: python/vyos/ifconfig/ethernet.py [hunk 1]
```diff
@@ -298,6 +311,11 @@ def update(self, config):
             duplex = config.get('duplex')
             self.set_speed_duplex(speed, duplex)
 
+        # Set interface ring buffers
+        if 'ring_buffers' in config.keys():
```
Comment (c-po) [inline python/vyos/ifconfig/ethernet.py]: please drop the optional .keys() call, its not necessary.

File: python/vyos/ifconfig/ethernet.py [hunk 2]
```diff
@@ -253,6 +253,19 @@ def set_ufo(self, state):
...
         return self.set_interface('ufo', state)
 
+    def set_ring_buffer(self, b_type, b_size):
+        """
+        Example:
+        >>> from vyos.ifconfig import EthernetIf
+        >>> i = EthernetIf('eth0')
+        >>> i.set_ring_buffer('rx', '4096')
+        """
+        cmd = '/sbin/ethtool -G {0} {1} {2}'.format(self.config['ifname'], b_type, b_size)
```
Comment (c-po) [inline python/vyos/ifconfig/ethernet.py:263]: please use self._cmd()

File: interface-definitions/interfaces-ethernet.xml.in [hunk 3]
```diff
@@ -268,6 +268,31 @@
...
+              <help>Shared buffers between the device driver and NIC</help>
+            </properties>
+            <children>
+              <leafNode name="rx">
+                <properties>
+                  <help>RX ring buffer</help>
+                  <valueHelp>
+                    <format>80-16384</format>
+                    <description>ring buffer size</description>
+                  </valueHelp>
```
Comment (c-po) [inline interface-definitions/interfaces-ethernet.xml.in:282]: Range validation is missing

**Review by c-po — CHANGES_REQUESTED** [review]

File: python/vyos/ifconfig/ethernet.py (hunk 2 above)
Comment (DmitriyEshenko) [reply python/vyos/ifconfig/ethernet.py:263]: Does not possible to use `self._cmd()` in this case, we need to check an exit code.

File: interface-definitions/interfaces-ethernet.xml.in [hunk 4]
```diff
@@ -268,6 +268,37 @@
             </properties>
             <defaultValue>auto</defaultValue>
           </leafNode>
+          <node name="ring-buffers">
```
Comment (c-po) [inline interface-definitions/interfaces-ethernet.xml.in]: Can we please use singular `ring-buffer` over plural `ring-buffers`? I guess this is what the CLI guideline says.

**Review by c-po — CHANGES_REQUESTED** [review]

File: python/vyos/ifconfig/ethernet.py (hunk 2 above)
Comment (c-po) [reply python/vyos/ifconfig/ethernet.py:263]: Got it thanks!

File: interface-definitions/interfaces-ethernet.xml.in (hunk 4 above)
Comment (DmitriyEshenko) [reply interface-definitions/interfaces-ethernet.xml.in]: Fixed.

**Pull request merged**